wordpress-mobile/WordPress-Android
# Jetpack Focus: Disable widgets during phase 4

Parent #17339 

This PR disables the "stats" widgets from showing when p4 is enabled. On app go-to-background and return-from-background the flag will be checked and the widgets will either be disabled or enabled. See `AppInitializer`.


Notes:  
- The "widgets" option remains, as there are other widgets beside stats.
- "Stats" shortcuts are still visible, they can not be removed dynamically. Tapping on a shortcut will be handled in a separate PR.

Before | After
-- | --
<img width="250" height="500" alt="Before" src="https://user-images.githubusercontent.com/506707/211614470-36bfaab3-313e-4bb9-82b4-d34cd48d7b33.png"> | <img width="250" height="500" alt="After" src="https://user-images.githubusercontent.com/506707/211614481-dfc2f192-81ae-457a-96cb-f28fd2727eb3.png">

**To test:**
- Install WP app
- Long press on the app icon until the shortcuts/widgets show
- Tap the widgets option
- ✅ Verify that the stats widgets are shown
- Login
- Navigate to -> Debug Settings -> and Enable `jp_removal_four` and restart the app
- Background the app
- Long press on the app icon until the shortcuts/widgets show
- Tap the widgets option
- ✅ Verify that the stats widgets are no longer shown

## Regression Notes
1. Potential unintended areas of impact
The widgets are enabled or disabled when they shouldn't be

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
